atom/browser/ui/win/thumbar_host.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/ui/win/thumbar_host.h"
+
+#include <shobjidl.h>
+
+#include "base/win/scoped_comptr.h"
+#include "base/win/win_util.h"
+#include "base/win/wrapped_window_proc.h"
+#include "base/strings/utf_string_conversions.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "ui/gfx/icon_util.h"
+#include "ui/gfx/win/hwnd_util.h"
+
+namespace atom {
+
+namespace {
+
+// The thumbnail toolbar has a maximum of seven buttons due to the limited room.
+const int kMaxButtonsCount = 7;
```

Comment:
I don't think we should limit the number of buttons on our size, if Windows isn't forbidding that we should not neither.